src/Raven.Server/Documents/Handlers/Debugging/DatabaseRecordHandler.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System.Threading.Tasks;
+using Raven.Client.ServerWide;
+using Raven.Server.Routing;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Web;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.Documents.Handlers.Debugging
+{
+    public class DatabaseRecordHandler : RequestHandler
+    {
+        internal static readonly string[] FieldsThatShouldBeExposedForDebug = new string[]
+        {
+            nameof(DatabaseRecord.DatabaseName),
+            nameof(DatabaseRecord.Encrypted),
+            nameof(DatabaseRecord.Disabled),
+            nameof(DatabaseRecord.EtagForBackup),
+            nameof(DatabaseRecord.DeletionInProgress),
+            nameof(DatabaseRecord.DatabaseState),
+            nameof(DatabaseRecord.Topology),
+            nameof(DatabaseRecord.ConflictSolverConfig),
+            nameof(DatabaseRecord.Sorters),
+            nameof(DatabaseRecord.Indexes),
+            nameof(DatabaseRecord.IndexesHistory),
+            nameof(DatabaseRecord.AutoIndexes),
+            nameof(DatabaseRecord.Settings),
+            nameof(DatabaseRecord.Revisions),
+            nameof(DatabaseRecord.RevisionsForConflicts),
+            nameof(DatabaseRecord.Expiration),
+            nameof(DatabaseRecord.Refresh),
+            nameof(DatabaseRecord.PeriodicBackups),
+            nameof(DatabaseRecord.ExternalReplications),
+            nameof(DatabaseRecord.SinkPullReplications),
+            nameof(DatabaseRecord.HubPullReplications),
+            nameof(DatabaseRecord.RavenEtls),
+            nameof(DatabaseRecord.SqlEtls),
+            nameof(DatabaseRecord.Client),
+            nameof(DatabaseRecord.Studio),
+            nameof(DatabaseRecord.TruncatedClusterTransactionCommandsCount),
+            nameof(DatabaseRecord.UnusedDatabaseIds),
+        };
+
+        [RavenAction("/debug/database-record", "GET", AuthorizationStatus.ValidUser, IsDebugInformationEndpoint = true)]
+        public Task GetDatabaseRecord()
```

Comment:
Let's avoid this EP completely and take it directly from the server

Reply:
Now a function instead of EP